backends-common/redis/src/main/java/org/apache/james/backends/redis/RedisHealthCheck.scala

```diff
@@ -62,46 +61,43 @@ class RedisHealthCheck @Inject()(redisConfiguration: RedisConfiguration, redisCl
       .onErrorResume(_ => SMono.just(Result.degraded(redisComponent, "Can not connect to Redis.")))
 
   @PreDestroy
-  def close(): Unit = healthcheckPerform.close()
+  def close(): Unit =
+    Mono.fromCompletionStage(rawRedisClient.shutdownAsync())
+      .subscribeOn(Schedulers.boundedElastic())
+      .subscribe()
 }
 
 sealed trait RedisHealthcheckPerform {
   def check(): SMono[Result]
-
-  def close(): Unit
 }
 
-class RedisStandaloneHealthCheckPerform(val redisConfiguration: StandaloneRedisConfiguration,
-                                        val redisClientFactory: RedisClientFactory,
-                                        val healthcheckTimeout: Duration) extends RedisHealthcheckPerform {
+class RedisRegularHealthCheckPerform(val redisClient: RedisClient, redisConfiguration: RedisConfiguration) extends RedisHealthcheckPerform {
```

Comment:
use `regular` to reflect that it is not cluster mode healthcheck seems not clear to me.

Reply:
Those classes literally disappear in the commit right after to just have only 1 healthcheck